adsrefpipe/models.py

```diff
@@ -0,0 +1,256 @@
+        self.source_filename = source_filename
+        self.source_modified = source_modified
+        self.status = status
+        self.date = date
+        self.total_ref = total_ref
+
+    def toJSON(self):
+        """
+        :return: values formatted as python dict, if no values found returns empty structure, not None
+        """
+        return {
+            'bibcode': self.bibcode,
+            'source_filename': self.source_filename,
+            'source_modified': self.source_modified,
+            'status': self.status,
+            'date': self.date,
+            'total_ref' : self.total_ref,
+        }
+
+
+class Resolved(Base):
+    __tablename__ = 'resolved'
+    history_id = Column(Integer, ForeignKey('history.id'), primary_key=True)
+    item_num = Column(Integer, primary_key=True)
+    reference_str = Column(String, primary_key=True)
+    bibcode = Column(String)
+    score = Column(Numeric)
+
+    def __init__(self, history_id, item_num, reference_str, bibcode, score):
+        """
+
+        :param history_id:
+        :param item_num
+        :param reference_str:
+        :param bibcode:
+        :param score:
+        """
+        self.history_id = history_id
+        self.item_num = item_num
+        self.reference_str = reference_str
+        self.bibcode = bibcode
+        self.score = score
+
+    def toJSON(self):
+        """
+        :return: values formatted as python dict, if no values found returns empty structure, not None
+        """
+        return {
+            'history_id': self.history_id,
+            'reference_str': self.reference_str,
+            'bibcode': self.bibcode,
+            'score': self.score,
+            'item_num': self.item_num
+        }
+
+
+class Compare(Base):
+    """
+    This table is for comparing classic resolver with service reference,
+    keeps track of service reference that matched classic reference
+    bibcode and score here is for classic
+
+    """
+    __tablename__ = 'compare'
+    history_id = Column(Integer, ForeignKey('history.id'), primary_key=True)
+    item_num = Column(Integer, primary_key=True)
+    bibcode = Column(String)
+    score = Column(Numeric)
+    state = Column(String)
+
+    def __init__(self, history_id, item_num, bibcode, score, state):
+        """
+
+        :param history_id:
+        :param item_num:
+        :param bibcode:
+        :param classic_score:
+        :param state:
+        """
+        self.history_id = history_id
+        self.item_num = item_num
+        self.bibcode = bibcode
+        self.score = score
+        self.state = state
+
+    def toJSON(self):
+        """
+        :return: values formatted as python dict, if no values found returns empty structure, not None
+        """
+        return {
+            'history_id': self.history_id,
+            'item_num': self.item_num,
+            'bibcode': self.bibcode,
+            'score': self.score,
+            'state': self.state,
+        }
+
+
+class XML(Base):
```

Comment:
Why only XML? I believe you may want to save any past reference (be it from XML, binary format, TEX or whatever else a parser can accept as a valid input)

Reply:
It is for only XML references. so here was my dilemma that I should get feedback from maybe Edwin and whoever else is going to check these. 
so in the table Resolved, there is a reference_str, https://github.com/golnazads/ADSReferencePipeline/blob/master/adsrefpipe/models.py#L153, for reprocessing of references, on the side of text, that should be enough to send it to the parser for processing and then resolving. On the side of XML I have been saving the a text reference of parsed xml reference, to be clear what the reference is. If we want to reprocess any XML references we need the original XML. On the side of classic, it uses the XML reference, but I thought it is more readable to have the text be available to the user in either text or xml and save these references in its own table. 
So I went back and forth which way to go and I went with more readable in Resolved, have an extra table for reprocessing. The other choice is have all original references in the Resolved table, whether readable or not.

Reply:
XML table removed.